### PR TITLE
Use `flex` for the back end gallery layout

### PR DESCRIPTION
**Pull request by fritzmg**
Due to subpixel rounding errors, gaps can appear in the back end gallery floating layout with certain image sizes and device pixel densities:

Screenshot 2024-08-31 113739

\
This PR fixes that by changing it to a flex layout instead.

### CODE REVIEW

**Comment by zoglo** [comment]
@fritzmg shouldn't we use the same approach as here (grid): https://github.com/contao/contao/pull/7072 ?

**Comment by fritzmg** [comment]
> @fritzmg shouldn't we use the same approach as here (grid): #7072 ?

What would be the advantage?

**Comment by zoglo** [comment]
Simply consistency, as it's basically the same issue. I wouldn't mind going with the flex variant: https://github.com/contao/contao/issues/7063#issuecomment-2023901262

**Comment by fritzmg** [comment]
The grid solution is unnecessarily complex for such cases and requires knowledge about the image size. Fixing #7063 with `flex` is also much simpler and would continue working if we change the preview image size, without having to change the CSS.

File: core-bundle/src/Resources/contao/themes/flexible/main.css [hunk 1]
```diff
@@ -1727,10 +1727,10 @@ ul.sgallery li {
 /* Misc */
 .ce_gallery ul {
 	overflow:hidden;
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/themes/flexible/main.css]: I think we can remove this. It was used to clear the floats of the `li` elements, which are no longer floating now.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**